### PR TITLE
NIFI-4988 Changed to handle an invalid ZIP file

### PR DESCRIPTION
**Pull request by apsaltis**
Changed the exception handling so that an invalid ZIP file is handled. When an invalid zip is processed, the exception is an IllegalArgumentException which was not being handled and thus the session was being rollbacked. 

Thank you for submitting a contribution to Apache NiFi.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [X ] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [X ] Does your PR title start with NIFI-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [X ] Has your PR been rebased against the latest commit within the target branch (typically master)?

- [ X] Is your initial contribution a single, squashed commit?

### For code changes:
- [ X] Have you ensured that the full suite of tests is executed via mvn -Pcontrib-check clean install at the root nifi folder?
- [X ] Have you written or updated unit tests to verify your changes?
- [N/A ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [N/A ] If applicable, have you updated the LICENSE file, including the main LICENSE file under nifi-assembly?
- [N/A ] If applicable, have you updated the NOTICE file, including the main NOTICE file found under nifi-assembly?
- [N/A ] If adding new Properties, have you added .displayName in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [N/A ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/test/java/org/apache/nifi/processors/standard/TestUnpackContent.java [hunk 1]
```diff
@@ -168,6 +168,39 @@ public void testZip() throws IOException {
...
+        unpackRunner.assertTransferCount(UnpackContent.REL_SUCCESS, 0);
+
+        autoUnpackRunner.assertTransferCount(UnpackContent.REL_FAILURE, 2);
+        autoUnpackRunner.assertTransferCount(UnpackContent.REL_ORIGINAL, 0);
+        autoUnpackRunner.assertTransferCount(UnpackContent.REL_SUCCESS, 0);
+
+        final List<MockFlowFile> unpacked = unpackRunner.getFlowFilesForRelationship(UnpackContent.REL_FAILURE);
+        for (final MockFlowFile flowFile : unpacked) {
+            final String filename = flowFile.getAttribute(CoreAttributes.FILENAME.key());
+           // final String folder = flowFile.getAttribute(CoreAttributes.PATH.key());
```
Comment (markap14) [inline nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/test/java/org/apache/nifi/processors/standard/TestUnpackContent.java]: Can probably remove this line.

**Comment by markap14** [comment]
I think I'm missing something obvious... but the PR adds a new file, invalid_data.zip but it doesn't add it to the RAT exclusions in the pom.xml... yet, it still pass the contrib-check. I'm not sure why.

**Comment by joewitt** [comment]
@markap14 i think often RAT can detect something is a binary file and not subject to having a license header.  It can't hurt to add the exclusion but i think it is normal that rat isn't unhappy about it.

**Comment by markap14** [comment]
Ah, ok, I didn't realize that @joewitt  - I thought it had to always be excluded. There are probably a handful of well-known magic headers or something then. Good to know!

**Comment by joewitt** [comment]
also possible that I am totally full of it.  One of these, possibly both, are true.

**Pull request closed without merging**